.github/workflows/detect-breaking-change.yml

```diff
@@ -1,5 +1,11 @@
 name: "Detect Breaking Changes"
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
```

Comment:
@dblock has raised a good point on the behavior of this workflow on `main` - maybe we should restrict this check to only pull requests.  @reta What do you think?

- [1] https://github.com/opensearch-project/OpenSearch/issues/13029